README.md

```diff
@@ -1,135 +1,268 @@
+new file formats and visualizing data from the corpus. The FAW supports
+various plugin types, which are further described in subsequent
+sections.
+
+### Parser Plugins
+
+The FAW supports parser plugins that validate file formats and extract
+format-specific features. These plugins execute external commands to
+parse files, capturing the exit code and standard output/error. The FAW
+allows mapping of these outputs to human-understandable features using
+regular expressions, with options for replacement and capture. The
+extracted features are stored in the database and can contribute to the
+valid/invalid decisions via the decision DSL mentioned in the previous
+section.
+
+TODO: Example ?
+
+### File-Detail Plugins
+
+File detail view plugins in the FAW visualize file contents by rendering
+them as HTML or other browser-displayable formats. Similar to parser
+plugins, they execute an external command that transforms the file
+content and streams it to the standard output. The FAW captures and
+renders this content in the browser based on the provided MIME type. An
+example of a file detail view plugin is PolyFile, which creates
+interactive HTML reports for various file formats, enabling users to
+view and search data structures within the file.
+
+### Corpus-Level Plugins
+
+Decision plugins in the FAW operate on the entire corpus, aggregating
+and visualizing information about dialects or common patterns. Like
+other plugins, they execute external commands, but can access the
+complete corpus and features captured during parser invocations.
+
+Results from the command are expected to be streamed in JSON line
+format, and the FAW can be optionally configured to capture and render
+custom HTML files generated by the plugin. This flexibility allows
+decision plugins to introduce new UI elements and actions within the
+FAW. FAW's root cause analysis are implemented as decision plugins.
+
+### Pipeline Plugins
+
+The FAW supports complex plugins with multiple stages and the capability
+to generate new data-driven parsers based on corpus inputs.
+
+TODO: Point to plugin tutorial
+
+### Transforming Inputs and Universal Output Parsers
+
+The FAW allows for input transformers that dynamically modify inputs
+before processing them. These transformers take a file from the corpus
+as input run external commands that produce transformed content on the
+standard output. Transformed inputs are treated as transient and not
+permanently stored. The SDF distribution, for example, use input
+transformers to generate a large corpus of SDF files from very simple
+seed input.
+
+The FAW also supports universal parser parsers, which operate on the
+output of all configured parsers, generating additional features. These
+parsers run external commands that accept specially formatted streams
+and emit JSON dictionaries of new features. Universal parser parsers
+unify format-specific considerations across all parsers
```

Comment:
No TODO here, but this sentence looks unfinished. Leaving a comment so this doesn't fall through the cracks

Reply:
Good catch. I have updated it slightly.